src/jtag3.c

```diff
@@ -1597,6 +1600,25 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
       }
     }
 
+    else if (str_starts(extended_param, "mode=") &&
+      (str_starts(pgmid, "pickit4") || str_starts(pgmid, "snap"))) {
+      // Flag a switch to AVR mode
+      if (str_caseeq(extended_param, "mode=avr")) {
+        PDATA(pgm)->pk4_snap_mode = PK4_SNAP_MODE_AVR;
+        continue;
+      }
+      // Flag a switch to PIC mode
+      else if (str_caseeq(extended_param, "mode=pic")) {
+        PDATA(pgm)->pk4_snap_mode = PK4_SNAP_MODE_PIC;
+        continue;
+      }
+      else {
```

Comment:
Don't else a continue. Below can be simplified to 
``` diff
diff --git a/src/jtag3.c b/src/jtag3.c
index 33bec590..e4af8cfc 100644
--- a/src/jtag3.c
+++ b/src/jtag3.c
@@ -1612,11 +1612,9 @@ static int jtag3_parseextparms(const PROGRAMMER *pgm, const LISTID extparms) {
         PDATA(pgm)->pk4_snap_mode = PK4_SNAP_MODE_PIC;
         continue;
       }
-      else {
-        pmsg_error("invalid mode setting '%s'. Use -xmode=avr or -xmode=pic\n", extended_param);
-        rv = -1;
-        break;
-      }
+      pmsg_error("invalid mode setting '%s'. Use -xmode=avr or -xmode=pic\n", extended_param);
+      rv = -1;
+      break;
     }
 
     else if (str_eq(extended_param, "help")) {
```

Actually. the else that follows the break doesn't need to be there as well.